### PR TITLE
chore: replace microbundle with rollup

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -126,16 +132,12 @@
...
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test --env=./custom-test-env.js",
     "test:e2e": "start-server-and-test 'serve build -l 3000' http://localhost:3000 'cypress run --record'",
-    "assets:generate": "yarn assets:svg:generate && yarn assets:font:generate",
-    "assets:svg:generate": "svgr -d src/lib/assets/svg --ext tsx --typescript src/lib/assets/svg && rm src/lib/assets/svg/index.tsx",
-    "assets:font:generate": "sass src/lib/assets/fonts/index.scss src/lib/assets/fonts/index.css --no-source-map -I node_modules",
-    "bundle": "microbundle --define process.env.REACT_APP_IS_WIDGET=true --tsconfig tsconfig.lib.json src/lib/index.tsx --format esm,cjs",
-    "bundle:depcheck": "node depcheck.js",
+    "bundle": "rollup --config --failAfterWarnings",
```
Comment (zzmp) [inline package.json:140]: NB: `--failAfterWarnings` replaces the existing `bundle:depcheck`.

File: tsconfig.lib.json [hunk 2]
```diff
@@ -1,15 +1,17 @@
 {
   "extends": "./tsconfig.base.json",
   "compilerOptions": {
-    "jsx": "preserve",
+    "jsx": "react-jsx",
```
Comment (zzmp) [inline tsconfig.lib.json:4]: NB: Using `preserve` (+ `acorn-jsx`) breaks usage; see https://github.com/rollup/rollup/issues/2822.

File: rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  ],
+  context: undefined,
+  external,
+  plugins: [
+    eslint({ include: ['**/*.js', '**/*.jsx', '**/*.ts', '**/*.tsx'] }),
+    json(), // imports json
+    replace({ ...replacements, preventAssignment: true }),
+    url(), // imports files (including svgs) as data URIs
+    svgr({ exportType: 'named', svgo: false }), // imports svgs as React components
+    sass(), // imports sass styles
```
Comment (justindomingue) [inline rollup.config.js:47]: do we need this?

File: package.json (hunk 1 above)
Comment (justindomingue) [inline package.json]: would probably rename this to `widgets:bundle` for readability

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import eslint from '@rollup/plugin-eslint'
+import json from '@rollup/plugin-json'
+import replace from '@rollup/plugin-replace'
+import url from '@rollup/plugin-url'
+import svgr from '@svgr/rollup'
+import sass from 'rollup-plugin-scss'
+import typescript from 'rollup-plugin-typescript2'
+
+import { dependencies } from './package.json'
```
Comment (justindomingue) [inline rollup.config.js]: i would add a top-level comment explaining why we are using rollup at the root of the interface project

Comment (justindomingue) [reply rollup.config.js]: could be confusing

File: rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+function external(source) {
+  const dep = deps.find((dep) => source === dep || source.startsWith(dep + '/'))
+  return Boolean(dep)
+}
+
+const config = {
+  input: 'src/lib/index.tsx',
+  output: [
+    {
+      file: 'dist/interface.js',
```
Comment (justindomingue) [inline rollup.config.js]: iiuc, this is the name of the output lib? should it be called something other than `interface`? `bundle`, `main`, `widget`, etc

Comment (justindomingue) [reply rollup.config.js]: I guess you're matching the metadata in `package.json`.. i think i would rename all but not a blocker

**Pull request merged**